crates/criticalup-core/src/errors.rs

```diff
@@ -70,6 +73,10 @@ pub enum Error {
         #[source]
         source: std::io::Error,
     },
+    #[error("Source error: {0}")]
```

Comment:
What is a "Source error"? What does this mean to the user? After fiddling around, it's an `io::Error` from `remove_dir_all`, so please instruct the user how to remedy if that fails.

```suggestion
    #[error("Could not remove directory {0}")]
    RemovingDirectory {
        path: PathBuf,
        #[source]
        source: std::io::Error,
    },
```